src/azure.ts

```diff
@@ -102,8 +120,211 @@ export async function setSubscriptionAsync(context: Context, subscription: strin
         return { succeeded: false, result: null, error: [sr.stderr] };
     }
 }
+export async function getClusterList(context: Context, subscription: string, clusterType: string) : Promise<ActionResult<ClusterInfo[]>> {
+    // log in
+    const login = await setSubscriptionAsync(context, subscription);
+    if (!login.succeeded) {
+        return {
+            actionDescription: 'logging into subscription',
+            result: { succeeded: false, result: [], error: login.error }
+        };
+    }
+
+    // list clusters
+    const clusters = await listClustersAsync(context, clusterType);
+    return {
+        actionDescription: 'listing clusters',
+        result: clusters
+    };
+}
+
+async function listClustersAsync(context: Context, clusterType: string) : Promise<Errorable<ClusterInfo[]>> {
+    let cmd = getListClustersCommand(context, clusterType);
+    const sr = await context.shell.exec(cmd);
+
+    if (sr.code === 0 && !sr.stderr) {
+        const clusters : ClusterInfo[] = JSON.parse(sr.stdout);
+        return { succeeded: true, result: clusters, error: [] };
+    } else {
+        return { succeeded: false, result: [], error: [sr.stderr] };
+    }
+}
+
+function listClustersFilter(clusterType: string): string {
+    if (clusterType == 'acs') {
+        return '?orchestratorProfile.orchestratorType==`Kubernetes`';
+    }
+    return '';
+}
+
+function getListClustersCommand(context: Context, clusterType: string) : string {
+    let filter = listClustersFilter(clusterType);
+    let query = `[${filter}].{name:name,resourceGroup:resourceGroup}`;
+    if (context.shell.isUnix()) {
+        query = `'${query}'`;
+    }
+    return `az ${getClusterCommand(clusterType)} list --query ${query} -ojson`;
+}
+
+async function listLocations(context: Context) : Promise<Errorable<Locations>> {
+    let query = "[].{name:name,displayName:displayName}";
+    if (context.shell.isUnix()) {
+        query = `'${query}'`;
+    }
+
+    const sr = await context.shell.exec(`az account list-locations --query ${query} -ojson`);
```

Comment:
`-ojson` -> `-o json`

Reply:
Is this a style preference?  I don't have strong feelings either way but have just developed the habit of `-ojson`.

Reply:
i take it as typos by mistake, it's OK for your usage.